src/server/pps/cmds/cmds.go

```diff
@@ -434,13 +442,52 @@ All jobs created by a pipeline will create commits in the pipeline's repo.
 			if err != nil {
 				return fmt.Errorf("error connecting to pachd: %v", err)
 			}
+
 			for {
 				request, err := cfgReader.NextCreatePipelineRequest()
 				if err == io.EOF {
 					break
 				} else if err != nil {
 					return err
 				}
+
+				// Add trace if env var is set
+				if _, ok := os.LookupEnv("PACH_TRACING_TARGET_REPO"); ok && tracing.IsActive() {
+					// unmarshal extended trace from RPC context
+					clientSpan, ctx := opentracing.StartSpanFromContext(
+						client.Ctx(),
+						"/pps.API/CreatePipeline",
+						ext.SpanKindRPCClient,
+						opentracing.Tag{string(ext.Component), "gRPC"},
+					)
+					defer clientSpan.Finish()
+					extendedTrace := extended.TraceProto{SerializedTrace: map[string]string{}} // init map
+					opentracing.GlobalTracer().Inject(
+						clientSpan.Context(),
+						opentracing.TextMap,
+						opentracing.TextMapCarrier(extendedTrace.SerializedTrace),
+					)
+					outputBranch := "master"
+					if request.OutputBranch != "" {
+						outputBranch = request.OutputBranch
+					}
+					extendedTrace.Branch = &pfs.Branch{
+						Repo: &pfs.Repo{Name: request.Pipeline.Name},
+						Name: outputBranch,
+					}
+					extendedTrace.Pipeline = request.Pipeline.Name
+					marshalledTrace, err := extendedTrace.Marshal()
+					if err == nil {
+						ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs(
+							extended.TraceCtxKey,
+							base64.URLEncoding.EncodeToString(marshalledTrace),
+						))
+						client = client.WithCtx(ctx)
+					} else {
+						fmt.Printf("ERROR marshalling extended trace proto: %v", err)
+					}
+				}
+
```

Comment:
Same as in `pfs/cmds` is this only for `list-job`? Or can you get it for other commands as well?